browser/test/Editor/NeovimEditor/SymbolsTests.ts

```diff
@@ -0,0 +1,206 @@
+            sinon.stub(menu, "onFilterTextChanged").get(() => onFilterTextChanged)
+            menuManager.create.returns(menu)
+
+            symbols["_requestSymbols"] = sinon.stub().resolves(["first symbol", "second symbol"])
+            const _symbolInfoToMenuItem = sinon.stub()
+            _symbolInfoToMenuItem.onCall(0).returns("first transformed")
+            _symbolInfoToMenuItem.onCall(1).returns("second transformed")
+            symbols["_symbolInfoToMenuItem"] = _symbolInfoToMenuItem
+        })
+
+        describe("openWorkspaceSymbolsMenu", () => {
+            let getKey: any
+
+            beforeEach(() => {
+                getKey = sinon.stub()
+                symbols["_getDetailFromSymbol"] = sinon.stub().returns(getKey)
+            })
+
+            it("requests workspace symbols when filter text is changed", async () => {
+                // setup
+                symbols.openWorkspaceSymbolsMenu()
+                clock.tick(30)
+                sinon.assert.notCalled(symbols["_requestSymbols"])
+
+                // action
+                onFilterTextChanged.dispatch("mock query")
+
+                // confirm
+                clock.tick(24)
+                sinon.assert.notCalled(symbols["_requestSymbols"])
+                clock.tick(1)
+                sinon.assert.calledWithExactly(
+                    symbols["_requestSymbols"],
+                    "mock buffer",
+                    "workspace/symbol",
+                    menu,
+                    { query: "mock query" },
+                )
+                await waitForPromiseResolution()
+                assertCommon()
+            })
+        })
+
+        describe("openDocumentSymbolsMenu", () => {
+            it("requests document symbols when completion menu is opened", async () => {
+                // action
+                await symbols.openDocumentSymbolsMenu()
+
+                // confirm
+                sinon.assert.calledWithExactly(
+                    symbols["_requestSymbols"],
+                    "mock buffer",
+                    "textDocument/documentSymbol",
+                    menu,
+                )
+                assertCommon()
+            })
+        })
+
+        const assertCommon = () => {
+            assert.deepEqual(symbols["_symbolInfoToMenuItem"].args, [
+                ["first symbol"],
+                ["second symbol"],
+            ])
+            sinon.assert.calledWithExactly(menu.setItems.lastCall, [
+                "first transformed",
+                "second transformed",
+            ])
+        }
+    }) // End describe open menus
+
+    describe("_requestSymbols", () => {
+        let menu: any
+        let buffer: any
+
+        beforeEach(() => {
+            menu = sinon.createStubInstance(Menu)
+            menu.isOpen.returns(true)
+            buffer = sinon.stub()
+            buffer.language = "mocklang"
+            buffer.filePath = "/mock/path"
```

Comment:
@feltech since our CI runs on multiple platforms you might need to use `path.join` here to make the mock paths platform agnostic